dtls_debug.c

```diff
@@ -127,31 +138,43 @@ dsrv_print_addr(const session_t *addr, char *buf, size_t len) {
       return 0;
 
     *p++ = '[';
+    --len;
 
     addrptr = &addr->addr.sin6.sin6_addr;
     port = ntohs(addr->addr.sin6.sin6_port);
 
     break;
   default:
-    memcpy(buf, "(unknown address type)", min(22, len));
-    return min(22, len);
+    // include terminating \000
+    append = min(23, len);
+    memcpy(p, "(unknown address type)", append);
+    if (len < 23) {
```

Comment:
I wonder why the default case needs zero-termination while the other cases may not be zero-terminated (and do not need to, hence the returned length).

Reply:
I consider the code to return a filled buffer only here, or after the snprintf.
And, maybe I'm wrong, I considered the snprintf to always terminate with 0.

Reply:
snprintf() does always include an explicit 0.  However, you cannot rely on the returned value as this could be oversize - it needs to be compared with the actual (length -1) provided to snprint() to prevent an overwrite of the provided buffer.

In the past, I have used strlen() to determine where the actual output has reached.

Reply:
Without overflow, it adds the 0. But with overflow, it's required to decide, if the output should fail, or the port number gets truncated. Anyway, I guess, I must check on all steps, if there is an overflow ...

Reply:
With overflow, it still adds the 0 as the last byte in the available buffer as indicated in (as well as in my experience of)

> The functions snprintf() and vsnprintf() do not write more than size bytes 
> (including the terminating null byte ('\0')).

Reply:
Maybe add in an assert() if there is insuffient space so that the buffer size issue can easily be caught (rather than just through code inspection) and the size adjusted accordingly.

Reply:
> With overflow, it still adds the 0 as the last byte

Hm, that was also, what I remembered. I try to check the docu, but maybe I still misunderstood it :-).

Reply:
> Maybe add in an assert() 

Not that bad. Then maybe, just at the start checking for the min size of the buffer?

Reply:
Works for me